app/controllers/topic_controller.rb

```diff
@@ -3,27 +3,21 @@
 require 'sinatra'
 require 'pony'
 require_relative '../clients/news_api'
+require './app/middleware/authenticator'
 
 # class TopicController
 class App < Sinatra::Application
   def fetch_user_choices_by_id(user_id)
     choices = []
-    user = User.find_by(id: user_id)
-    p user.subjects.all.empty?
-    redirect '/categories' if user.subjects.all.empty?
-    user.subjects.all.each do |item|
+    redirect '/categories' if @user.subjects.all.empty?
```

Comment:
Why do we have `.subjects.all`? I think the `.all` is not necessary because `subjects` is plural